homework/minieieva_daria/report.md

```diff
@@ -0,0 +1,100 @@
+### Week 6 Regular Expressions Homework
+
+### This is a template for the report of 21 problems on [this site](http://regextutorials.com/excercise.html).
+
+### Fork this repository, change this file and send us the link for grading.
+
+### We recommend you (re)watch our presentation, and use a [cheatsheet](./cheatsheet.md) while working on the following problems.
+
+### If you are having problems with figuring out the correct solution, skip it for later, and if you are completely lost, use the hints and try to understand them.
+
+---
+
+### Problem 1: Floating point numbers
+```
+\d\.\d+
+```
+### Problem 2: Years before 1990
+```
+\d\s\w*\s\w*[,|:]*(\s\w*)*\(1[0-9][0-8][0-9]\)
+```
+### Problem 3: Hexadecimal colors
+```
+#([A-Z]|[0-9]){6}
+```
+### Problem 4: Grayscale colors
+```
+#((([A-z]|[0-9]){1,2})\2{2})
+```
+### Problem 5: Too long lines
+```
+.{30,}
+```
+### Problem 6: Remove repeating words
+```
+(["]*\w+[-']*\w*)[\s.",]\1[\s.",]
+$1 
+```
+### Problem 7: Match HTML tags
+```
+<[!/]*\w*\s*\w*>
+```
+### Problem 8: Cut numbers two digits after floating point
+```
+(\d\.\d{2})\d*
+```
+### Problem 9: Digit commas formatting
+```
+insert your pattern here
+```
+### Problem 10: Match lowercase function declarations
+```
+function+\s[a-z]\w*\(\w*\)
```

Comment:
You don't need a `+` after the `function` here for sure. And also, theoretically, a function could have a lot of parameters specified with spaces included, your pattern does not account for that.

Once again, that's me just being an asshole and thinking whether there is something to improve here, but your pattern is nice already!